vector/src/main/java/im/vector/app/features/userdirectory/KnownUsersFragment.kt

```diff
@@ -107,6 +120,47 @@ class KnownUsersFragment @Inject constructor(
         }
     }
 
+    private fun setupAddByQrCodeView() {
+        val qrStartForActivityResult = registerStartForActivityResult { activityResult ->
+            if (activityResult.resultCode == Activity.RESULT_OK) {
+                val result = QrCodeScannerActivity.getResultText(activityResult.data)!!
+
+                // Latter condition to be replaced by URL(result).protocol == "matrix" when MSC2312 is complete
+                if (result.isValidUrl() && URL(result).host == "matrix.to") {
+                    // Remove query string parameters and initial slash from URL().ref to get MXID
+                    val mxid = URL(result).ref.split("?")[0].drop(1)
+
+                    val existingDm = session.getExistingDirectRoomWithUser(mxid)
+
+                    if (existingDm == null) {
+                        // The following assumes MXIDs are case insensitive. Change if needed
+                        if (mxid.toLowerCase(Locale.getDefault()) != session.myUserId.toLowerCase(Locale.getDefault())) {
```

Comment:
it's better to write `if (mxid.equals(other = session.myUserId, ignoreCase = true))` and deal with the equal case first, to avoid `!` in the condition, if we have a `if` and a `else` block

Reply:
done